trie.h

```diff
@@ -27,13 +27,25 @@ typedef struct
     uint8_t     completion_len;     // length of completion string
     uint8_t     num_backspaces;     // number of backspaces to send before the completion string
     uint8_t     func_code;          // special function code
-    uint8_t     context_match_len;
 } st_trie_payload_t;
 
-bool st_trie_get_completion(st_trie_t *trie, st_key_buffer_t *search, st_trie_payload_t *res);
+typedef struct
+{
+    uint16_t            trie_match_index;
+    uint8_t             context_match_len;
```

Comment:
since this is new code, let's use the new naming convention now: seq_match_len

Reply:
oh and seq_match_len should be type int I think.

Reply:
We don't do any math on `seq_match_len`, so it was probably safe, but I'm 100% in favor of using `int` for all lengths and indexes that aren't in an array or trie.